Added Keeper (#95)
sidestream-tech/unified-auctions-ui: bot-twitter/README.md

# Auctions twitter bot

> ## What is our "Unified Auctions Bot"?

Reviewer: ```suggestion
## What is "Unified Auctions Bot"?
```

> Our "Unified Auctions Bot", is a server side application, with a collection of tools 
> for interacting with the blockchain and participating in MakerDAO collateral Auctions. 
> The bot can be hosted by any user, provided, they configure the bot properly over the environment variables.
> 
> The bot currently has two main features. The first is the ability to tweet about new auctions, quickly alerting
> users about new auctions. We are hosting our own version of this Twitter Bot. You can find its account [here](https://twitter.com/MakerDaiBot).
> 
> The second feature is the "Keeper". This is an automated system for bidding on certain auctions. Through configuration in the environment variables, you can 
> provide a wallet secret key as well as a minimum profitability (in DAI). Once this profitability threshold is reached
> the bot will automatically bid on the Auction for you.
> 
> In order to use the bot please fill out all required [environment variables](#environment-variables) below. You can skip all optional variables, or those
> that are not required for the feature you want to use. Afterwards follow our Setup guides.

Reviewer: ```suggestion
"Unified Auctions Bot" is a server-side application with two distinct functionalities:

- `Twitter bot` – a script that periodically fetches collateral auctions and publish a tweet if it finds a new auction
- `Keeper bot` – a script that periodically fetches collateral auctions, and their market values, and if find that any of them are profitable, executes respective transactions (authorisations and bidding)

The bot can be hosted by any user, provided, they configure related environment variables. For more details, please refer to [`Environment variables`](#environment-variables) section below.
```

- `REFETCH_INTERVAL`: (optional, default 60 seconds) – interval between auction fetching requests
- `FRONTEND_ORIGIN`: (required) An origin to which the bot will redirect users (valid example: `https://auctions.makerdao.network`)
- `WALLET_PRIVATE_KEY`: (required for server side execution) The secret key of the wallet (https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key)

> - `MIN_PROFIT_DAI`: (required for keeper) The minimum amount of profit an auction must yield before the keeper automatically bids on it

Reviewer: Let's prepend both of those keys with `KEEPER_`

## Environment variables

In order to configure you bot, you must create a `.env` file in the root of `/bot-twitter`.
Below you can find a list of all available environment variables, along with an explanation of 

> what each variable determines. If not all parts of the bot are needed, you can leave some variables undefined.

Reviewer: `.env` file is only one option of how to provide env variables. I would not include this text here since `Development Setup` already explains it

- `INFURA_PROJECT_ID`: (required) [infura](https://infura.io/) project id (can be found in: dashboard -> ethereum -> create new project -> settings -> keys). Note: this project can not be restricted by the origin.
- `ETHEREUM_NETWORK`: (optional, default `kovan`) – internal network name on which the bot poll for auctions. Available options can be found in [constants/NETWORKS](../core/src/constants/NETWORKS.ts)
- `REFETCH_INTERVAL`: (optional, default 60 seconds) – interval between auction fetching requests
- `FRONTEND_ORIGIN`: (required) An origin to which the bot will redirect users (valid example: `https://auctions.makerdao.network`)
- `KEEPER_*` variables - a set of set for the keeper

>     - `KEEPER_WALLET_PRIVATE_KEY`: (required for server side execution) The secret key of the wallet (https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key)
>     - `KEEPER_MIN_PROFIT_DAI`: (required for keeper) The minimum amount of profit an auction must yield before the keeper automatically bids on it

Reviewer: ```suggestion
    - `KEEPER_WALLET_PRIVATE_KEY`: (required) The wallet private key (https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key)
    - `KEEPER_MIN_PROFIT_DAI`: (required) The minimum amount of DAI an auction must yield before the keeper automatically bids on it
```
